fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -1880,6 +1915,60 @@ public static Builder newBuilder() {
             return new Builder();
         }
 
+        /**
+         * The desired action if the index is disabled.
+         * @return requested {@link DesiredAction}
+         */
+        public DesiredAction getIfDisabled() {
+            return ifDisabled;
+        }
+
+        /**
+         * The desired action if the index is in write only mode.
+         * @return requested {@link DesiredAction}
+         */
+        public DesiredAction getIfWriteOnly() {
+            return ifWriteOnly;
+        }
+
+        /**
+         * The desired action if the index is in partly built, but the requested policy mismatches the previous one.
+         * @return requested {@link DesiredAction}
+         */
+        public DesiredAction getIfMismatchPrev() {
+            return ifMismatchPrev;
+        }
+
+        /**
+         * The desired action if the index is in readable mode (i.e. already built).
+         * @return requested {@link DesiredAction}
+         */
+        public DesiredAction getIfReadable() {
+            return ifReadable;
+        }
+
+        /**
+         * Get the appropriate desired action for a given index state.
+         * @param state A given index state
+         * @return The appropriate {@link DesiredAction}
+         */
+        public DesiredAction getStateDesiredAction(IndexState state) {
+            switch (state) {
+                case DISABLED:      return getIfDisabled();
+                case WRITE_ONLY:    return getIfWriteOnly();
+                case READABLE:      return getIfReadable();
+                default: throw new RecordCoreException("bad index state: ", state);
+            }
+        }
+
+        /**
+         * Enforce an index rebuild when the index state is WRITE_ONLY. This is useful when other conditions
+         * require a full rebuild.
+         */
+        public void enforceRebuildIfWriteOnly() {
+            ifWriteOnly = DesiredAction.REBUILD;
```

Comment:
Hm, I would rather that the policy remain immutable, if possible. In particular, I'm concerned about users who may cache a copy of their favorite `IndexingPolicy` and not expect for the policy object itself to sometimes mutate.

Reply:
That's a very good point! 
I'll replace all modifications of this object with a copy/modify

Reply:
Fixed